main.js

```diff
@@ -1,163 +1,277 @@
+        options: {
+            wifi: false,
+            speech: false,
+            hdr: false,
+            bluetooth: false,
+            ambiLight: false,
+        },
+        originalStock: 10,
+        sold: 8,
     },
-    originalStock: 10,
-    sold: 8,
-  },
-  {
-    type: '32WL1A63DG',
-    name: 'HD TV',
-    brand: 'Toshiba',
-    price: 161,
-    availableSizes: [32],
-    refreshRate: 50,
-    screenType: 'LED',
-    screenQuality: 'Full HD',
-    smartTv: false,
-    options: {
-      wifi: false,
-      speech: false,
-      hdr: true,
-      bluetooth: false,
-      ambiLight: false,
+    {
+        type: '32WL1A63DG',
+        name: 'HD TV',
+        brand: 'Toshiba',
+        price: 161,
+        availableSizes: [32],
+        refreshRate: 50,
+        screenType: 'LED',
+        screenQuality: 'Full HD',
+        smartTv: false,
+        options: {
+            wifi: false,
+            speech: false,
+            hdr: true,
+            bluetooth: false,
+            ambiLight: false,
+        },
+        originalStock: 10,
+        sold: 8,
     },
-    originalStock: 10,
-    sold: 8,
-  },
-];
\ No newline at end of file
+];
+//1a
+const tvsStockMapper = inventory.map((stock) => {
+    return stock.originalStock - stock.sold
+});
+const totalTvsInStock = () => {
+    let allStock = 0
+    for (let i = 0; i < inventory.length; i++) {
+        allStock = allStock + tvsStockMapper[i];
+    }
+    return allStock
+}
+// 1b
+const stockField = document.getElementById('current-stock');
+stockField.innerText = `Totale huidige voorraad: \n${totalTvsInStock().toString()} stuks.`
+//  2a
+const tvTypeMapper = inventory.map((type) => {
+    return type.type;
+});
+//  2b
+const soldOut = inventory.filter((sold) => {
+    return sold.originalStock - sold.sold === 0;
+});
+//  2c
+const ambiLight = inventory.filter((ambi) => {
+    return ambi.options.ambiLight;
+});
+//  2d (aka 2e)
+const priceSort = inventory.sort((a, b) => a.price - b.price);
+// 3a
+const price = inventory.map((prijs) => {
+    return prijs.price;
+});
+let totalRevenue = 0;
+for (let i = 0; i < inventory.length; i++) {
+    totalRevenue += tvsStockMapper[i] * price[i];
+}
+const target = document.getElementById('goal-revenue');
+target.innerText = `Verkoopwaarde huidige voorraad: € ${totalRevenue.toString()},-`;
+//3b(aka 6b)
+const sold = inventory.map((verk) => {
+    return verk.sold;
+});
+let totalRevenueSoFar = 0;
+for (let i = 0; i < inventory.length; i++) {
+    totalRevenueSoFar += sold[i] * price[i];
+}
+const made = document.getElementById('current-revenue');
+made.innerText = `Verkoopwaarde verkochte voorraad: € ${totalRevenueSoFar.toString()},-`;
+
+// 4
+const tvOne = document.getElementById('tv1');
+tvOne.innerText = `Tv typenummer 1: ${tvTypeMapper[0].toString()}.`;
+
+const tvTwo = document.getElementById('tv2');
+tvTwo.innerText = `Tv typenummer 2: ${tvTypeMapper[4].toString()}.`;
+
+// 5a
+const brandMapper = inventory.map((merk) => {
```

Comment:
Qua naamgeving moet je het in het Engels schrijven, dus niet `merk`, maar `brand`.